honeybee_3dm/face.py

```diff
@@ -0,0 +1,112 @@
+    rhino3dm_file = rhino3dm.File3dm.Read(path)
+    tolerance = rhino3dm_file.Settings.ModelAbsoluteTolerance
+
+    hb_faces = []
+
+    layer_dict = {
+        layer.Name: layer.Index for layer in rhino3dm_file.Layers}
+
+    # A Layer dictionary with layer name : (Honeybee face_type, Class) structure
+    layer_to_hb_object = {
+        'roof': (face_types.roof_ceiling, Face),
+        'wall': (face_types.wall, Face),
+        'floor': (face_types.floor, Face),
+        'airwall': (face_types.air_boundary, Face),
+        'shade': (None, Shade),
+        'aperture': (None, Aperture),
+        'door': (None, Door)}
+
+    for layer in layer_to_hb_object.keys():
+
+        if layer in layer_dict.keys():
+            hb_face_type, hb_face_module = layer_to_hb_object[layer]
+
+            # Gathering planar geometries from the layer
+            rhino_faces = [object for object in rhino3dm_file.Objects
+                           if object.Attributes.LayerIndex == layer_dict[layer]]
+
+            # Creating face names
+            face_names = [geo.Attributes.Name if geo.Attributes.Name else layer +
+                          str(uuid.uuid4())[:8] for geo in rhino_faces]
+
+            # for each rhino geometry gathered, Converting the Rhino3dm geometry
+            # into a Ladybug Face3D objects
+            for count, face in enumerate(rhino_faces):
+                rh_face = face.Geometry
+
+                if rh_face.ObjectType == rhino3dm.ObjectType.Brep:
+                    lb_face = brep_to_face3d(rh_face)
+
+                elif rh_face.ObjectType == rhino3dm.ObjectType.Extrusion:
+                    lb_face = extrusion_to_face3d(rh_face)
+
+                elif rh_face.ObjectType == rhino3dm.ObjectType.Mesh:
+                    lb_face = mesh_to_face3d(rh_face)
+                else:
+                    print(
+                        f'There are objects on layer "{layer}" that this library does not support.')
+
+                # Converting Ladybug Face3D into Honeybee Objects
+                for face_obj in lb_face:
+
+                    if hb_face_module == Face:
+                        hb_face = hb_face_module(clean_and_id_string('{}_{}'.format(
+                            face_names[count], count)), face_obj, hb_face_type)
+
+                    elif hb_face_module == Shade:
+                        hb_face = hb_face_module(clean_and_id_string(
+                            '{}_{}'.format(face_names[count], count)), face_obj)
+
+                    elif hb_face_module == Aperture:
+                        hb_face = hb_face_module(clean_and_id_string(
+                            '{}_{}'.format(face_names[count], count)), face_obj)
+
+                    elif hb_face_module == Door:
+                        hb_face = hb_face_module(clean_and_id_string(
+                            '{}_{}'.format(face_names[count], count)), face_obj)
+                    else:
+                        pass
```

Comment:
I assume since you have already checked the types up there we should never get here and you can remove this.

Reply:
Removed